api/src/main/java/org/apache/cloudstack/api/command/user/vm/DeployVMCmd.java

```diff
@@ -293,7 +292,8 @@ public Long getDomainId() {
             }
         }
         if (ApiConstants.BootType.UEFI.equals(getBootType())) {
```

Comment:
@shwstppr I think, no need to add (empty) boot type to custom parameters when boot mode is not specified.

```suggestion
        if (getBootMode() != null && ApiConstants.BootType.UEFI.equals(getBootType())) {
```

Reply:
Not sure abt this as in UI we allow empty boot mode and if we are skipping here, we will have to handle UI better by not allowing empty boot mode when UEFI is selected as boot type. Otherwise, user will expect VM to boot with UEFI just by changing the boot type.
@rhtyd @nvazquez @davidjumani any comments on the behaviour?
I feel maybe we should just handle the NPE in 4.15 and refactor boot options handling in UI & API in a separate PR for 4.16

Reply:
@sureshanaparti ^^

Reply:
Suresh's comment makes sense, kindly discuss and conclude @shwstppr

Reply:
@sureshanaparti as discussed offline, I've refactored changes to throw exception when bootmode is empty for boottype=UEFI